### PR TITLE
Add ping endpoint

### PR DESCRIPTION
**Pull request by dmcgowan**
Ping endpoint added so clients can get the list of authentication challenges before making an initial request.

The clients could be coded without a ping endpoint but that requires closer integration of the token auth with the client which does not currently exist. This is the fastest approach to get authentication functional.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -93,3 +95,14 @@ func Run(ctx context.Context, addr, tlsCertFile, tlsKeyFile string, trust signed
 
 	return err
 }
+
+func apiBase(ctx context.Context, w http.ResponseWriter, r *http.Request) *errors.HTTPError {
```
Comment (endophage) [inline server/server.go]: Can we put this in the handlers package somewhere for consistency.

**Comment by diogomonica** [comment]
@endophage can you take a look again?

**Comment by dmcgowan** [comment]
ping @endophage

**Comment by endophage** [comment]
My error branch will change how the error is returned, but I'll deal with that when I rebase that branch so LGTM.

**Pull request merged**